Layout Tweaks

samdeane opened the pull request:
Added an alternative status layout.
Added the option to hide user handles.

The post/reply/etc controls appear on rollover in the alt layout, which is a little jarring, especially when scrolling. I tried animating it but there are some glitches with the image views (which resize weirdly before animating), which makes that look a bit ugly too. 

I think the rollover state could probably be debounced/delayed slightly so that you had to hover for a few seconds, which would clean up scrolling. The controls could also be overlaid and offset from the bottom of the message, so that they don’t affect the message layout. Or the controls could show when the message is selected, rather than when rolling over (better on iOS too perhaps).

samdeane committed:
added an alt row type

samdeane committed:
Show original poster's avatar & name

samdeane committed:
option to hide poster's handle

samdeane committed:
don't need quoted by argument

samdeane committed:
better relative date formatting

samdeane committed:
nicer avatar styling

samdeane committed:
try to optimise layout for detail

samdeane committed:
try to optimise layout for detail

samdeane committed:
improved layout

schwa committed:
Use SwiftSoup.

samdeane committed:
disabled the animation, which doesn't work as expected right now

samdeane committed:
generalise relative dates

The .relative style in Text is accurate to the second, which seemed a little overkill - especially since it updated every second which was visually jarring

samdeane committed:
generalise relative dates

samdeane committed:
Optionally hide account handles

samdeane committed:
tweaked layout

samdeane committed:
Circular avatar style

samdeane committed:
reverted obsolete renaming

schwa committed:
Use SwiftSoup for HTML parsing.

schwa commented:
Looks like this includes the relative date PR too? Can you rebase that out?

Also a screenshot would be nice. Cheers!

schwa committed:
Remove HTML2Markdown.

schwa committed:
Update README.

schwa committed:
Update README.md

schwa committed:
Fix failing unit tests.

schwa committed:
Fixes #33 - fail gracefully with bad blurhash.

schwa committed:
Fixes #34. Add quick filter/search function.

schwa committed:
Fixes #13 Dead space at top of timeline was hidden progressview.

schwa committed:
Partial attempt at moving settings to main navigation stack on iOS.

schwa committed:
Fixes #36 put toolbar options into popover on iOS

schwa committed:
Fixes #37 refresh task for iOS.

schwa committed:
Improve status row for iPhones.

Get NewPost mostly working for iPhones. Start on photo picker button.

schwa committed:
Update packages.

schwa committed:
Use Photo picker and content type woes.

schwa committed:
Flesh out Notifications tab a little.

schwa committed:
Added a quick "file bug" button.

schwa committed:
Add AccountActions in the spirit of StatusActions.

schwa committed:
Disable crashing updateAccount.

samdeane commented:
This branch is my work-in-progress branch, so probably shouldn't be merged as-is. 

As discussed in #3, I will split out individual self-contained changes and submit them individually.

samdeane commented:
I've switched this PR to a draft, but close it down if that works better for you.

schwa committed:
Get rid of instances list - we can easily bring it back later.

schwa committed:
Minor refactoring and reorg around Timeline.

schwa committed:
Fixes #29. Persist timelines.

Fixes #8. No longer spamming relationships.

Access timelines and relationships through Service Channels.

schwa committed:
Prevent issues with zero size card images.

schwa committed:
More work on #8 - cleaning up relationship spam.

schwa committed:
Tidy & lint

schwa committed:
Cleanup warnings.

schwa committed:
Attempt to get relationships view workign so we can debug relationships.

schwa committed:
WIP

schwa committed:
Tidy & lint

schwa committed:
Fix data race.

schwa committed:
Fix concurrency warnings.

schwa committed:
Fix more warnings.

schwa committed:
Warning cleanup

schwa committed:
More warning cleanup.

schwa committed:
Update to latest everything to get rid of some deprecation warnings.

schwa committed:
Initial Announcements impl.

schwa committed:
Lock down swiftlint rules some more (https://github.com/schwa/swiftlint-autodetect/blob/main/README.markdown)

schwa committed:
swiftlint cleanup

schwa committed:
Fix relationship spam issue #8

Rehook up unfollow/disable reposts/get follow unfollow button working in notifications view.

schwa committed:
Add ticket info.

schwa committed:
Fixes #67 - clean up nav stack code.

schwa committed:
Make Storage registration only available at init time.

schwa committed:
Fixes #24 make Storage an actor.

schwa committed:
Create CHANGELOG.md

schwa committed:
Update README.md

schwa committed:
Deprecated Stored & DataStorage.

schwa committed:
Fixes #22 - deprecate Stored instead of re-working it.

schwa committed:
Fixes #71 - open Storage at init time.

schwa committed:
WIP

schwa committed:
Refactor mainview _again_ and add ConsoleLog.

schwa committed:
Placeholder Account page.

schwa committed:
Add quicklook for media.

schwa committed:
Quick accessibility pass. Add button & popover for image descriptions.

schwa committed:
Model update.

GuyEnglish committed:
Add a basic image description generator for alt text.

GuyEnglish committed:
Add `ImageDescription` struct to generate place-holder alt text.

schwa committed:
Working on account views.

schwa committed:
Placeholder for bookmarks.

GuyEnglish committed:
Add ImageDescription

schwa committed:
Bookmark view first pass.

schwa committed:
Clean up as many errors as possible.

schwa committed:
Oops. Status should not be hashable.

schwa committed:
Fix some of the breakage i just inflicted on the code.

schwa committed:
Merge pull request #74 from GuyEnglish/image_desc

Add ImageDescription

schwa committed:
Sanitize Everything url.

schwa committed:
Work on image uploads.

schwa committed:
Update Everythng.

schwa committed:
Sanitize Everything.git url

schwa committed:
WIP

schwa committed:
Get auto-generated media descriptions (mostly) working.

schwa committed:
Fix compilation issues on iOS.

samdeane committed:
Merge branch 'layout-tweaks' into layout-tweaks-integration

# Conflicts:
#	MastodonAgain.xcodeproj/project.pbxproj
#	MastodonAgain/Content/LargeStatusRow.swift
#	MastodonAgain/Content/StatusRow.swift
#	MastodonAgain/Content/TimelineView.swift
#	MastodonAgain/Main/MainView.swift
#	MastodonAgain/Support/Support.swift

samdeane committed:
Re-integrated some of my changes with the latest integration branch.
